docs/devcontainerjson-reference.md

```diff
@@ -0,0 +1,172 @@
+| `service` | string | **Required** when [using Docker Compose](https://code.visualstudio.com/docs/remote/create-dev-container#_use-a-dockerfile). The name of the service `devcontainer.json` supporting services / tools should connect to once running.  |
+| `runServices` | array | An array of services in your Docker Compose configuration that should be started by `devcontainer.json` supporting services / tools. These will also be stopped when you disconnect unless `"shutdownAction"` is `"none"`. Defaults to all services. |
+| `workspaceFolder` | string | Sets the default path that `devcontainer.json` supporting services / tools should open when connecting to the container (which is often the path to a volume mount where the source code can be found in the container). Defaults to `"/"`. |
+
+## General devcontainer.json properties
+
+| Property | Type | Description |
+|----------|------|-------------|
+| `name` | string | A name for the dev container displayed in the UI. |
+| `forwardPorts` | array | An array of port numbers or `"host:port"` values  (e.g. `[3000, "db:5432"]`) that should always be forwarded from inside the primary container to the local machine (including on the web for Codespaces). The property is most useful for forwarding ports that cannot be auto-forwarded because the related process that starts before the `devcontainer.json` supporting service / tool connects or for forwarding a service not in the primary container in Docker Compose scenarios (e.g. `"db:5432"`). Defaults to `[]`. <br /><br /> ⚠️ Codespaces does not yet support the `"host:port"` variation of this property. |
+| `portsAttributes` | object | Object that maps a port number, `"host:port"` value, range, or regular expression to a set of default options. See [port attributes](#port-attributes) for available options. For example: <br />`"portsAttributes": {"3000": {"label": "Application port"}}` <br /><br /> ⚠️ Codespaces does not yet support the `"host:port"` variation of this property.|
+| `otherPortsAttributes` | object | Default options for ports, port ranges, and hosts that aren't configured using `portsAttributes`. See [port attributes](#port-attributes) for available options. For example: <br /> `"otherPortsAttributes": {"onAutoForward": "silent"}` |
+| `remoteEnv` | object | A set of name-value pairs that sets or overrides environment variables for the `devcontainer.json` supporting service / tool (or sub-processes like terminals) but not the container as a whole. Environment and [pre-defined variables](#variables-in-devcontainerjson) may be referenced in the values. Be sure **Terminal > Integrated: Inherit Env** is checked in settings or the variables will not appear in the terminal. For example: <br />`"remoteEnv": { "PATH": "${containerEnv:PATH}:/some/other/path", "MY_VARIABLE": "${localEnv:MY_VARIABLE}" }`<br />Updates are applied when the `devcontainer.json` supporting service / tool is restarted (or the window is reloaded) |
```

Comment:
There's a VS Code specific setting listed here we should probably remove.

Reply:
Removed the sentence.